src/services/FulfillmentService.js

```diff
@@ -49,6 +55,47 @@ class FulfillmentService {
       throw err;
     }
   };
+
+  closeTheLoop = async (event, context, callback) => {
+    context.callbackWaitsForEmptyEventLoop = false;
+
+    try {
+      const now = new Date();
+      const today = new Date(now.getFullYear(), now.getMonth(), now.getDate());
+      const followUps = await FollowUp.find({ delivery_date: today }).exec();
+
+      const emailLog = new EmailMessageLog({
```

Comment:
You should be saving an email log for each follow-up email, not just one for all of them. The point of this it to provide an audit log for each email that was sent out. If you only create a single log record, then all the subsequent updates to the record will override each other. The initial request, in this case, should be set to the follow-up document that you pulled from mondgodb for each follow-up.

Reply:
In sendEmail function we get emailLog._id parameter and set it to emailLogId parameter for sqs.sendMessage(params). But in closeTheLoop we have batch of emails parameters. As i see we need to send arrays of these ids

Reply:
And we also need to process these ids in generateEmails function

Reply:
We need to post multiple messages to the queue, not a single message with multiple ids.

Reply:
done